Tasks/UseDotNetV2/globaljsonfetcher.ts

```diff
@@ -0,0 +1,62 @@
+import * as fileSystem from "fs";
+import { DotNetCoreVersionFetcher } from "./versionfetcher";
+import { VersionInfo } from "./models";
+import * as taskLib from 'azure-pipelines-task-lib/task';
+
+export class globalJsonFetcher {
+
+    private workingDirectory: string;
+    private versionFetcher: DotNetCoreVersionFetcher = new DotNetCoreVersionFetcher();
+    /**
+     * The global json fetcher provider functionality to extract the version information from all global json in the working directory.
+     * @param workingDirectory 
+     */
+    constructor(workingDirectory: string) {
+        this.workingDirectory = workingDirectory;
+    }
+
+    /**
+     * Get all version information from all global.json starting from the working directory.
+     * @param packageType define if you want install the sdk or the runtime.
+     */
+    public async Get(packageType: string): Promise<VersionInfo[]>{
+        var versionInformation: VersionInfo[] = new Array<VersionInfo>();
+        for (let index = 0; index < this.getVersionStrings().length; index++) {
+            const version = this.getVersionStrings()[index];
+            // TODO: find out if include preview versions do anything wrong here.
+            var versionInfo = await this.versionFetcher.getVersionInfo(version, packageType, true); 
+            versionInformation.push(versionInfo);
+        }
+        return versionInformation
+            .sort((a,b) => a.getVersion().localeCompare(b.getVersion()));        
+    }
+
+    private getVersionStrings(): string[]{
+        let filePathsToGlobalJson = taskLib.findMatch(this.workingDirectory, "**/*global.json");
+        if (filePathsToGlobalJson == null || filePathsToGlobalJson.length == 0) {
+            throw taskLib.loc("FailedToFindGlobalJson");
+        }
+        return filePathsToGlobalJson.map(path => {
+            return this.readGlobalJson(path).sdk.version;
```

Comment:
map puts empty objects in array when condition is not present. This will happen when we handle empty global.json. you might want to think about that

Reply:
see the conversation here:

https://github.com/microsoft/azure-pipelines-tasks/pull/10734/files/7f2dc6001c34abf6f66b6954cee8d1a281462542#r298030574